tasks/configure.yml

```diff
@@ -17,13 +22,13 @@
   shell: pg_dropcluster --stop {{postgresql_version}} {{postgresql_cluster_name}}
   sudo: yes
   sudo_user: "{{ postgresql_service_user }}"
-  when: postgresql_cluster_reset and pgdata_dir_exist.changed
+  when: postgresql_cluster_reset and pgdata_dir_exist.changed and pgconf_dir.stat.isdir is defined
 
 - name: PostgreSQL | Reset the cluster - create a new one (with specified encoding and locale)
   shell: pg_createcluster --start --locale {{postgresql_locale}} -e {{postgresql_encoding}} -d {{postgresql_data_directory}} {{postgresql_version}} {{postgresql_cluster_name}}
   sudo: yes
   sudo_user: "{{ postgresql_service_user }}"
-  when: postgresql_cluster_reset and pgdata_dir_exist.changed
+  when: postgresql_cluster_reset and pgdata_dir_exist.changed and not pgconf_dir.stat.isdir is defined
```

Comment:
You are right, I didn't have much experience with stat, I'll change it and squash It right away!